### DISCUSSION TITLE
Have all ACE/TAO releases (or micro releases) been tagged as a Git tag in the Git repository?

### DISCUSSION THREAD

**Discussion by pierre-rouleau** [DOCGroup/ACE_TAO#2265] [Q&A] [answered]
Hi,

I'm trying to understand the Git branches and Git tag that when also looking at the [ACE_TAO releases](https://github.com/DOCGroup/ACE_TAO/releases) and [ACE_TAO tags](https://github.com/DOCGroup/ACE_TAO/tags)  so that I can checkout a specific version from my clone of the Git repo.

For example, let's say an old version like [ACE+TAO-7_0_11](https://github.com/DOCGroup/ACE_TAO/releases/tag/ACE%2BTAO-7_0_11).  

- I was expecting that this version would be identified as one of your Git repo tags.  Is it the case?  I did not see it.  
- Should I simply just use the Git hash for the commit of this release (in the case of [ACE_TAO 7.0.11](https://github.com/DOCGroup/ACE_TAO/commit/ec326c77c9a3bb7335dc6e97b9b4e2009efe2adb) it would be:

    `git checkout ec326c77c9a3bb7335dc6e97b9b4e2009efe2adb`  

which, on a system I use, leads to:

``` bash
 
bash$ git co ec326c77c9a3bb7335dc6e97b9b4e2009efe2adb
Note: switching to 'ec326c77c9a3bb7335dc6e97b9b4e2009efe2adb'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at ec326c77c9a ACE+TAO-7_0_11
```

**Comment by mitza-oci** [comment]
I see the tag for `ACE+TAO-7_0_11` which points to commit ec326c77c9a3bb7335dc6e97b9b4e2009efe2adb

**Reply by pierre-rouleau** [reply]
Thanks.  I see it too in Github, but when I list the branches from my local clone I have not found any other way than doing a checkout from using the hash digest number.  I am most probably missing something.   Should I be able to see '7_0_11" on a branch name somewhere?

**Reply by mitza-oci** [reply, marked as answer]
I have it in my local clone.

```
$ git show ACE+TAO-7_0_11
tag ACE+TAO-7_0_11
Tagger: Johnny Willemsen <jwillemsen@remedy.nl>
Date:   Mon Dec 19 12:03:34 2022 +0100

ACE+TAO-7_0_11

commit ec326c77c9a3bb7335dc6e97b9b4e2009efe2adb (tag: ACE+TAO-7_0_11)
Author: Johnny Willemsen <jwillemsen@remedy.nl>
Date:   Mon Dec 19 12:02:51 2022 +0100

    ACE+TAO-7_0_11
[...]
```

**Reply by pierre-rouleau** [reply]
OK, I see that too.  But if I use `git branch --all` I don't see it.  I was trying to see the list of tags that correspond to branches that have been published.    

Now I realize the repo is organized by tags and we should use the following:

- Use `git tag` to list all ACE/TAO release points
- Use `git checkout tags/ACE+TAO-7_0_1` to checkout a specific version, in this case 7_0_1.

In retrospect I should have asked that: **do all releases have branches or only tags?**  The answer seems to be: "tags"

**Reply by mitza-oci** [reply]
I don't think "tags that correspond to branches that have been published" is a concept that git supports.  However, I think the default behavior for fetch (and clone) is that tags which refer to commits that are fetched are also fetched.  So you will have all the tags that are relevant, along with others.

**Reply by pierre-rouleau** [reply]
Thanks, you are correct.  Too many VCS around...
